src/game.cpp

```diff
@@ -2959,12 +2965,14 @@ void Game::toggleFreeMove(float *statustext_time)
 	static const wchar_t *msg[] = { L"free_move disabled", L"free_move enabled" };
 
 	bool free_move = !g_settings->getBool("free_move");
-	g_settings->set("free_move", bool_to_cstr(free_move));
 
 	*statustext_time = 0;
 	statustext = msg[free_move];
-	if (free_move && !client->checkPrivilege("fly"))
+	if (free_move && !client->checkPrivilege("fly")) {
 		statustext += L" (note: no 'fly' privilege)";
+	} else {
```

Comment:
This part is probably too invasive and a confusing change. I think I'll undo this one. (and the same for noclip)